controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -186,13 +188,13 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(ctx context.Context
 	if err != nil {
 		errmsg := fmt.Errorf("error reconciling NodeNetworkConfigurationPolicy at desired state apply: %s, %v", nmstateOutput, err)
 
-		enactmentConditions.NotifyFailedToConfigure(errmsg)
+		defer enactmentConditions.NotifyFailedToConfigure(errmsg)
```

Comment:
This is dangerous here since we have the other defer `defer policyconditions.Update(r.Client, request.NamespacedName)` we don't know which one will be run first and nncp conditions update depends on enactments since it's calculate from there, I suggest to just move this just before return and add a comment.

Reply:
I'm okay with dropping the `defer` here, as most of the time in which killing handler casues inconsistency is spent in Progressing state.
I would just like to point out that `deferred` calls are stacked and upon return executed in last-in-first-out order https://tour.golang.org/flowcontrol/13